### PR TITLE
Add validation to agent name

### PR DESCRIPTION
**Pull request by Tostti**
# Description
This PR aims to add a validation to the agent name, in order to avoid entering incorrect names that may cause errors in different places of the app.

# Issues Resolved
https://github.com/wazuh/wazuh-kibana-app/issues/4927

# Evidence 
image
image
image

# Test
**Scenario:** have a Wazuh environment and navigate to agents/deploy new agent
**When** the users enter a name with [invalid characters](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/client.html#agent-name)
**Then** it should show an error telling that, and not show the deploy commands

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: public/controllers/agent/components/register-agent.js [hunk 1]
```diff
@@ -293,10 +298,18 @@ export const RegisterAgent = withErrorBoundary(
 
     setAgentName(event) {
       this.setState({ agentName: event.target.value });
-    }
-
-    setAgentName(event) {
-      this.setState({ agentName: event.target.value });
+      if (/^[a-z0-9-_.]+$/i.test(event.target.value) || event.target.value.length <= 0) {
```
Comment (asteriscos) [inline public/controllers/agent/components/register-agent.js:301]: Nitpick: If you are going to use the same regular expression more than once, you could store it in a constant to improve maintainability.

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark: 
Test: :heavy_check_mark: 

<details>

<summary> Evidence </summary>

image

image


</details>

File: public/controllers/agent/components/register-agent.js (hunk 1 above)
Comment (Tostti) [reply public/controllers/agent/components/register-agent.js]: Solved in https://github.com/wazuh/wazuh-kibana-app/pull/5021/commits/72fcfabb864a08850d43795d989c8cf97909c488

**Review by asteriscos — APPROVED** [review]

**Review by Desvelao — APPROVED** [review]
**review**

Code :green_circle: 

Tests
| Scenario | Result |
| --- | --- |
| 1 | :green_circle: |
| 2 | :green_circle: |

<details>
  <summary>Scenario 1 :green_circle: </summary>
  
 image

</details>

<details>
  <summary>Scenario 2 :green_circle: </summary>

image  
</details>

**Pull request merged**